irc/client.go

```diff
@@ -185,9 +185,11 @@ func (client *Client) run() {
 	var msg ircmsg.IrcMessage
 
 	defer func() {
-		if r := recover(); r != nil {
-			client.server.logger.Error("internal",
-				fmt.Sprintf("Client caused panic, disconnecting: %v\n%s", r, debug.Stack()))
+		if client.server.RecoverFromErrors() {
+			if r := recover(); r != nil {
+				client.server.logger.Error("internal",
+					fmt.Sprintf("Client caused panic, disconnecting: %v\n%s", r, debug.Stack()))
```

Comment:
Hmm... could we do something along the lines of:

1. If RecoverFromErrors is true, recover and log the error.
2. If RecoverFromErrors is false, log the error and then let it kill the server.

I'd sorta like to always know what caused the server to fail, even if we choose to kill the server anyways right after. Maybe if we can't re-raise the error, we could just log the stack as above, then shutdown the server manually?